app/controllers/runtime/spaces_controller.rb

```diff
@@ -212,11 +211,14 @@ def delete_isolation_segment(guid)
 
     def before_update(obj)
       if request_attrs['isolation_segment_guid']
-        raise CloudController::Errors::ApiError.new_from_details('NotAuthorized') unless SecurityContext.admin?
+        check_isolation_segment_access!(obj)
 
-        isolation_segment = IsolationSegmentModel[guid: request_attrs['isolation_segment_guid']]
-        raise CloudController::Errors::ApiError.new_from_details('InvalidRelation',
-                "Could not find Isolation Segment with guid: #{request_attrs['isolation_segment_guid']}") unless isolation_segment
+        isolation_segment_guids = obj.organization.isolation_segment_models.map(&:guid)
```

Comment:
Not sure what you mean - organization.isolation_segment_models is the accessor for ISes from orgs...?  The space needs to navigate to the ISes to which the owning org is entitled, I don't really see a shorter path to get there?